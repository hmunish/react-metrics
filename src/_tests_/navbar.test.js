import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Navbar from '../components/navbar';

const mockStore = configureStore([]);

describe('testing navbar component', () => {
  it('Navbar should render in the document', () => {
    const mockCurrencies = {
      currenciesData: [],
      curCountryCode: '',
      isLoading: false,
      isError: false,
    };

    const store = mockStore({
      currency: mockCurrencies,
    });

    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    );
    const bar = getByTestId('nav-test'); // eslint-disable-next-line no-unused-expressions
    expect(bar).toBeInTheDocument;
  });

  it('should render correct snapshot', () => {
    const mockCurrencies = {
      currenciesData: [],
      curCountryCode: '',
      isLoading: false,
      isError: false,
    };

    const store = mockStore({
      currency: mockCurrencies,
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
