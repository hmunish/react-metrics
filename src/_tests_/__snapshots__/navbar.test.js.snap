// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing navbar component should render correct snapshot 1`] = `
<nav
  data-testid="nav-test"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <li
      id="back"
    />
  </a>
  <p>
    All country currencies
  </p>
  <ul>
    <li
      id="mic"
    />
    <li
      id="settings"
    />
  </ul>
</nav>
`;
